homeassistant/components/input_datetime/reproduce_state.py

```diff
@@ -31,18 +31,12 @@ def is_valid_datetime(string: str) -> bool:
 
 def is_valid_date(string: str) -> bool:
     """Test if string dt is a valid date."""
-    try:
-        return dt_util.parse_date(string) is not None
-    except ValueError:
-        return False
+    return dt_util.parse_date(string) is not None
```

Comment:
This was my idea as well in the first place. But in fact we can rely on the underlying methods working well. Furthermore we test invalid values already using `"not_supported"` and a date that refers to the 25th month.

Reply:
Since you're removing two catches of `ValueError` exceptions, I think your tests should include calls that try to hit these removed lines and thus demonstrate that they aren't necessary. `"not_supported"` will reach the `except ValueError:` block in `dt_util.parse_date`, so that's covered. But the `except ValueError:` block in `dt_util.parse_time` does not get reached in these tests. `"invalid:date:time"` would hit both. It's also more obvious than `"2010-25-10"`, which looks a lot like `"2010-10-25"`.

Reply:
So you are suggesting that I should replace both test cases with `"invalid:date:time"`?

Reply:
I think 3 examples would be ideal, 1 for each of the `is_valid_*` functions in `reproduce_state.py`. Something like:

* `"not-valid-date"`
* `"not:valid:time"`
* `"1234-56-78 90:12:34"`

Reply:
Done. Thanks